### ISSUE TITLE
Dropdown (PF4): When the first DropdownItem in the Dropdown has an onClick handler keyboard navigation does not work

### ISSUE THREAD

**Issue by KKoukiou** [patternfly/patternfly-react#3737]
As the title says - when the first item in the DropdownItems has an onClick handler that will consume the space key down on the dropdown element, which is supposed to expand the dropdown, and the keyboard navigation will  not work properly.

Reproducer is the following code:
``` jsx
import React from 'react';
import {
  Dropdown,
  DropdownToggle,
  DropdownItem,
  DropdownSeparator,
  DropdownPosition,
  DropdownDirection,
  KebabToggle
} from '@patternfly/react-core';
import { ThIcon } from '@patternfly/react-icons';

class PositionRightDropdown extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
    this.onToggle = isOpen => {
      this.setState({
        isOpen
      });
    };
    this.onSelect = event => {
      this.setState({
        isOpen: !this.state.isOpen
      });
      this.onFocus();
    };
    this.onFocus = () => {
      const element = document.getElementById('toggle-id-5');
      element.focus();
    };
  }

  render() {
    const { isOpen } = this.state;
    const dropdownItems = [
      <DropdownItem key="test" component="button" onClick={() => console.info('First link pressed')}>Link</DropdownItem>
    ];
    return (
      <Dropdown
        onSelect={this.onSelect}
        position={DropdownPosition.right}
        toggle={<DropdownToggle onToggle={this.onToggle}>Dropdown</DropdownToggle>}
        isOpen={isOpen}
        dropdownItems={dropdownItems}
      />
    );
  }
}
```

**Comment by KKoukiou** [comment]
Actually it's not when oClick handler is used but when component="button" is used in the dirst dropdownItem

**Comment by rachael-phillips** [comment]
Hi @KKoukiou we've seen this and will follow up soon with a response.

**Comment by kmcfaul** [comment]
@KKoukiou 
I've taken the snippet and played around with it [here](https://codesandbox.io/s/dropdown-bug-svd4m), but I'm having trouble reproducing an issue with keyboard navigation. What specifically is not working properly for you?

For reference, I've adjusted the snippet to include the ID on the dropdown toggle and added a couple more items to test the up/down arrows.

**Comment by KKoukiou** [comment]
@kmcfaul thanks for the codesandbox - it makes it very easy to show you the problem now. Basically move the focus with the keyboard to the dropdown element. Make sure the element has focus but it's not expanded. Now pressing 'space' should expand the element and leave it expanded. See the accessibility guidelines here for example https://webaim.org/techniques/keyboard/ .
On this sandbox this does not happen, instead the dropdown expands and closes immediately.

**Comment by kmcfaul** [comment]
Ah I'm seeing it now @KKoukiou, on Firefox specifically. It appears that Firefox is triggering two events in a single space keypress (triggering onKeyDown in the toggle and then onClick in the first item, which then closes the menu). We will have to catch the `onKeyUp` event in both components to prevent this extra trigger.

**Comment by KKoukiou** [comment]
@tlabaj please reopen and as discussed in the previous cockpit/PF4 migration meeting please make sure that the bot does not close actual bugs without human approval.
